src/platforms/common/aux_serial.c

```diff
@@ -40,14 +40,14 @@ static uint32_t aux_serial_active_baud_rate;
 
 static char aux_serial_receive_buffer[AUX_UART_BUFFER_SIZE];
 /* Fifo in pointer, writes assumed to be atomic, should be only incremented within RX ISR */
-static uint8_t aux_serial_receive_write_index = 0;
+static uint32_t aux_serial_receive_write_index = 0;
```

Comment:
Given how this expands static memory usage for SRAM, what's the justification here for going to uint32_t instead of uint16_t please?

Reply:
My technical argument for this is when you're targeting a natively 32-bit ALU machine, such as Cortex-M3 ARMv7-M (with 16-bit instructions Thumb1), a 32-bit unsigned integer type has the fastest executing codegen. 16-bit or 8-bit involves a halfword or byte load/store and masking: uxth, uxtb. Needs Compiler Explorer for hard evidence. Also consider using atomics, ldrex/strex will happily operate on 32-bit variables. Oh, and the AHB is 32-bit, where the SRAM is, accesses steal a couple cycles for AHB handshaking anyways. 
+12 bytes is not much to pay for correctness. If you want me to make it 6 bytes plus masking, then I'll need to retest with uint16_t.

Reply:
> 16-bit or 8-bit involves a halfword or byte load/store and masking: uxth, uxtb.

We'd expect `ldrh` and `ldrb` for those loads, not the masking instructions. This should make things atomic even with uint16_t's.

Reply:
You're right, loads and stores are equivalent. It's the increment that bothers me (and %=SIZE). I'll temporarily accept using uint16_t.
Thinking some more, an F723 at 216M Hclk with UART set up for DIV18, that is 12 Mbaud, will have to perform the buffer switch in 18 * 10 baud = 180 clocks of its CM7; for optimistic DIV12 (in OVER8, 18Mbaud), in 12 * 10 = 120 clocks, or about half a microsecond, to avoid DMA spilling into the next half of buffer. If code requires more overhead, then this should be stated in platform readme and I won't care anymore.
Similarly for F411 at 96M Hclk and UART at DIV16, so 6Mbaud, there's 16 * 10baud = 160 clocks to do this for FS.